x/supplier/keeper/supplier.go

```diff
@@ -133,6 +133,31 @@ func (k Keeper) hydrateSupplierServiceConfigs(ctx context.Context, supplier *sha
 	supplier.Services = supplier.GetActiveServiceConfigs(currentHeight)
 }
 
+// hydrateDehydratedSupplierServiceConfigs populates a supplier with its service configurations
+// but excludes service_config_history and rev_share details to reduce response size.
+//
+// The function:
+// - Determines which configurations are active at the current block height
+// - Sets the supplier's active services without rev_share information
+// - Does not populate service_config_history
+func (k Keeper) hydrateDehydratedSupplierServiceConfigs(ctx context.Context, supplier *sharedtypes.Supplier) {
```

Comment:
Not a big fan of the method name.

What about: `hydrateSupplierWithLightweightServices` or `hydrateSupplierWithActiveServiceConfigs`